src/library/scala/collection/Iterable.scala

```diff
@@ -912,3 +895,60 @@ object Iterable extends IterableFactory.Delegate[Iterable](immutable.Iterable) {
 
 /** Explicit instantiation of the `Iterable` trait to reduce class file size in subclasses. */
 abstract class AbstractIterable[+A] extends Iterable[A]
+  with IterableFactoryDefaults[A @uncheckedVariance, Iterable]
+
+trait IterableFactoryDefaults[A, +CC[x] <: IterableOps[x, CC, CC[x]]] extends IterableOps[A, CC, CC[A]] {
```

Comment:
A scaladoc comment explaining the purpose of this trait would be welcome.

Reply:
true, will do!